### ISSUE TITLE
twisted.names.client.Resolver.__init__.__doc__ error

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11545]
|<img alt="Nerdbeard's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Nerdbeard reported|
|-|-|
|Trac ID|trac#1622|
|Type|defect|
|Created|2006-04-05 20:36:15Z|

The docs indicate this should work:

```
        server_address = "10.0.0.1"
        resolver = Resolver(servers=[server_address])
```

But what is required is actually:

```
        server_address = "10.0.0.1"
        port = 53
        resolver = Resolver(servers=[(server_address, port)])
```

<details><summary>Searchable metadata</summary>

```
trac-id__1622 1622
type__defect defect
reporter__Nerdbeard Nerdbeard
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__names names
keywords__documentation documentation
time__1144269375000000 1144269375000000
changetime__1215701430000000 1215701430000000
version__ 
owner__ 
cc__Nerdbeard cc__exarkun cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

It seems pretty clear to me, but how about this change:

```
Index: twisted/names/client.py
===================================================================
--- twisted/names/client.py     (revision 16546)
+++ twisted/names/client.py     (working copy)
@@ -53,9 +53,10 @@
         for modification and re-parsed if it is noticed to have changed.
 
         @type servers: C{list} of C{(str, int)} or C{None}
-        @param servers: If not None, interpreted as a list of addresses of
-        domain name servers to attempt to use for this lookup.  Addresses
-        should be in dotted-quad form.  If specified, overrides C{resolv}.
+        @param servers: If not None, interpreted as a list of IPv4 TCP or
+        UDP addresses of domain name servers to attempt to use for this
+        lookup.  Addresses should be in dotted-quad form.  If specified,
+        overrides C{resolv}.
 
         @type resolv: C{str}
         @param resolv: Filename to read and parse as a resolver(5)
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to Nerdbeard|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Nerdbeard's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Nerdbeard commented|
|-|-|

You know, I totally overlooked the @type field.  It would have been clear to me if I had read it.  Sorry.

But as long as I've got a ticket going, I might as well suggest a change:

```
        @param servers: If not None, interpreted as a list of (host,port) pairs
        specifying addresses of domain name servers to attempt to use for this
        lookup.  Hosts should be in dotted-quad form.  If specified,
        overrides C{resolv}.
```

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra removed owner|
|-|-|

Can someone review the suggested doc update, the doc update makes sense.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @thijstriemstra|
|-|-|

Please apply.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24257]) Apply Resolver.__doc__ patch: update docstring for the servers parameter

Author: Nerdbeard
Reviewer: exarkun, thijs, therve
Fixes: [#11545](https://github.com/twisted/twisted/issues/11545)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
